### ISSUE TITLE
DateTime Kind is lost during serialization via BinaryFormatter

### ISSUE THREAD

**Issue by LadislavLang** [dotnet/corefx#33461]
```c#
    using System;
    using System.IO;
    using System.Runtime.Serialization.Formatters.Binary;

    class Program
    {
        static void Main(string[] args)
        {
            var input = new[]
            {
                new DateTime(2018, 11, 13, 0, 0, 0, DateTimeKind.Local),
                new DateTime(2018, 11, 13, 0, 0, 0, DateTimeKind.Unspecified),
                new DateTime(2018, 11, 13, 0, 0, 0, DateTimeKind.Utc)
            };

            var formatter = new BinaryFormatter();
            using (var stream = new MemoryStream())
            {
                formatter.Serialize(stream, input);

                stream.Seek(0, SeekOrigin.Begin);

                var output = (DateTime[])formatter.Deserialize(stream);

                // Output:
                // NetFramework (4.7):  Local, Unspecified, Utc
                // CoreApp (2.1):       Unspecified, Unspecified, Unspecified

                Console.WriteLine($"{output[0].Kind}, {output[1].Kind}, {output[2].Kind}");
            }

            Console.ReadKey();
        }
    }
```

**Comment by stephentoub** [comment]
The problem is here:
https://github.com/dotnet/corefx/blob/6e55e0fa6711a15e1e02a6948532979a142c993b/src/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs#L82

**Comment by laurentiudiaconu** [comment]
Hi gents,

Which release of .Net Core 2.1 was this change included in?

Thanks,
Laurentiu

**Comment by ViktorHofer** [comment]
The mentioned fix isn't part of .NET Core 2.1 but 3.0. You can try it out with the recently released preview-2 version.

**Comment by laurentiudiaconu** [comment]
Is it not going to be included in 2.1 at all?

I'm working in an enterprise environment and am afraid I can't use a preview version.

Thanks

**Comment by ViktorHofer** [comment]
cc @danmosemsft
